src/radical/pilot/agent/agent_0.py

```diff
@@ -839,6 +839,9 @@ def _prepare_env(self, env_name, env_spec):
 
         # if the ve lives outside of the pilot sandbox, link it
         if path:
+            # expand any env variables in path
+            path, _, _ = ru.sh_callout('echo -n "%s"' % path, shell=True)
+            path = os.path.abspath(path)
```

Comment:
This expansion either needs documentation in `pilot.prepare_env`, or we should remove it again.  If we keep it, then error checks should be added to the code.